### PR TITLE
zed: rename Column type to Field

### PR DESCRIPTION
**Pull request by nwt**
The changes outside package zed are as limited as possible, leaving a similar renaming of identifiers in those packages for later.

### CODE REVIEW

File: value.go [hunk 1]
```diff
@@ -215,14 +215,14 @@ func (r *Value) nth(column int) zcode.Bytes {
 	return zv
 }
 
-func (r *Value) Columns() []Column {
-	return TypeRecordOf(r.Type).Columns
+func (r *Value) Fields() []Field {
+	return TypeRecordOf(r.Type).Fields
 }
 
 func (v *Value) DerefByColumn(col int) *Value {
```
Comment (mattnibs) [inline value.go:222]: Should this be renamed `DerefByField`?

File: recordbuilder.go [hunk 2]
```diff
@@ -189,10 +189,10 @@ func (r *RecordBuilder) Encode() (zcode.Bytes, error) {
 // TypedColumns takes an array of Types for the individual fields
 // and constructs an array of Columns that reflects the fullly
 // typed structure.  This is suitable for e.g. allocating a descriptor.
```
Comment (mattnibs) [inline recordbuilder.go:191]:
```suggestion
// Fields takes an array of Types for the individual fields
// and constructs an array of Fields that reflects the fullly
// typed structure.  This is suitable for e.g. allocating a descriptor.
```

**Review by mattnibs — APPROVED** [review]

File: value.go (hunk 1 above)
Comment (nwt) [reply value.go:222]: Indeed.

File: recordbuilder.go (hunk 2 above)
Comment (nwt) [reply recordbuilder.go:191]: Good catch!

File: value.go (hunk 1 above)
Comment (nwt) [reply value.go:222]: On second thought, I think this should become DerefByIndex or FieldByIndex, so I'll leave it as is and we'll sort out the name in another PR.

**Pull request merged**